### PR TITLE
fix(prow/jenkins-operator): fix log fetching for jenkins job in folder

### PR DESCRIPTION
**Pull request by wuhuizuo**
the plain console log for job in a folder is `/job/folder/job/jobname/<buildNum>/consoleText`

### CODE REVIEW

**Comment by wuhuizuo** [comment]
@petr-muller 😃 expect your reviewing

**Comment by petr-muller** [comment]
sorry I only have limited capacity for reviews and so I mostly review parts that I'm familiar with (=not jenkins-operator, unfortunately)

**Comment by wuhuizuo** [comment]
> sorry I only have limited capacity for reviews and so I mostly review parts that I'm familiar with (=not jenkins-operator, unfortunately)

@petr-muller  could you help me to assign a reviewer? thanks! 🤝 

Currently no reviewers in `OWNERS` file in `jenkins-operator` or `jenkins`, maybe we should add some reviewers and approvers for `jenkins*`

File: prow/cmd/jenkins-operator/logs_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import "testing"
+
+func Test_getRealJenkinsLogPath(t *testing.T) {
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name    string
+		args    args
```
Comment (petr-muller) [inline prow/cmd/jenkins-operator/logs_test.go:27]:
```suggestion
		path    string
```
... and get rid of the unnecessary struct

File: prow/cmd/jenkins-operator/logs.go [hunk 2]
```diff
@@ -56,3 +66,18 @@ func handleLog(jc *jenkins.Client) http.HandlerFunc {
 		}
 	}
 }
+
+func getRealJenkinsLogPath(path string) (string, error) {
+	jobMatches := reJenkinsJobURL.FindStringSubmatch(path)
+	if len(jobMatches) == 0 {
```
Comment (petr-muller) [inline prow/cmd/jenkins-operator/logs.go:72]: Here we should check that we have exactly 5 items, not just non-emptiness

File: prow/cmd/jenkins-operator/logs_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			args: args{path: "/job/abc/1/consoleText"},
+			want: "/job/abc/1/consoleText",
+		},
+		{
+			name: "nested job",
+			args: args{path: "job/folder-l1/foleer-L2/the-job/1/consoleText"},
+			want: "job/folder-l1/job/foleer-L2/job/the-job/1/consoleText",
+		},
+		{
+			name: "nested job with lead splash",
```
Comment (petr-muller) [inline prow/cmd/jenkins-operator/logs_test.go:42]:
```suggestion
			name: "nested job with lead slash",
```

File: prow/cmd/jenkins-operator/logs.go [hunk 4]
```diff
@@ -56,3 +66,18 @@ func handleLog(jc *jenkins.Client) http.HandlerFunc {
...
+		return "", errors.New("job name invalid")
```
Comment (petr-muller) [inline prow/cmd/jenkins-operator/logs.go:73]: pls make the error more informative, something like "job name does not match <regex>"

**Review by petr-muller — APPROVED** [review]
/shrug
/meow

Thanks!

**Pull request merged**